packages/material-ui-lab/src/Pagination/usePagination.js

```diff
@@ -0,0 +1,142 @@
+import { useControlled } from '@material-ui/core/utils';
+
+export default function usePagination(props = {}) {
+  const {
+    boundaryCount: boundaryCountProp = 1,
+    componentName = 'usePagination',
+    count = 1,
+    defaultPage = 1,
+    disabled = false,
+    hideNextButton = false,
+    hidePrevButton = false,
+    onChange: handleChange,
+    page: pageProp,
+    showFirstButton = false,
+    showLastButton = false,
+    siblingCount = 1,
+    ...other
+  } = props;
+
+  // TODO: Update all formulae to remove this adjustment
```

Comment:
Subject of another PR?

Reply:
I tried twice and gave up. It should be hard, but apparently it is for me :D

Reply:
Well :smile: 

> There are 2 hard problems in computer science: cache invalidation, naming things, and off-by-1 errors.